Лабораторная работа_6

ChervonenkoDaniil opened the pull request:
Возникли трудности с реализацией task4, описано в комментариях внутри кода

ChervonenkoDaniil committed:
Лабораторная работа_6

Возникли трудности с реализацией task4, описано в комментариях внутри кода

SemendyaevRodion commented on Лабораторная работа_6/task4.py:
```diff
@@ -0,0 +1,24 @@
+# Очень долго думал, но не придумал, зачем функция воспринимает аргумент разделителя, но, по заданию добавил
+def csv_to_list_dict(filename, delimiter=',', new_line='\n') -> list[dict]:
+    # Получаем данные, форматируем для дальнейшей работы
+    with open(filename, 'r') as f:
```
Решение вышло громоздким. Можно его упростить. Когда читается файл, то внутри этого файла находится курсор, который перемещается по файлу. То есть можно сначала прочитать первую строчку файла f.readline(), это будет хедер (название столбцов). Как только была прочитана первая строчка курсор автоматический перейдет на вторую, с этого места вы уже можете использовать list comprehension, куда помещаете функцию zip(), которая перебирает одновременно два списка (значения из хедера и значения из строки файла). Так как вы должны создать словарь, то вы используете преобразование в dict().

SemendyaevRodion commented on Лабораторная работа_6/task4.py:
```diff
@@ -0,0 +1,24 @@
+        headers = f.readline().split(delimiter)
+        data = f.readlines()
+        data = [i.split(delimiter) for i in data]
+    # Вероятнее всего, для удаления из конца строки, но не понял, как реализовать
```
все очень спроса вы можете использовать срез [:-1], чтобы избавиться от последнего элемента.

ChervonenkoDaniil committed:
Update task4.py

Проблема заключалась в удалении именно \n в конце последней строки списка, внедрил отдельные строки с удалением внутрь строки с импортом данных
